test/Garnet.test/RespHashTests.cs

```diff
@@ -693,7 +693,163 @@ public void CheckHashOperationsOnWrongTypeObjectSE()
             RespTestsUtils.CheckCommandOnWrongTypeObjectSE(() => db.HashStringLength(keys[0], hashFields[0][0]));
         }
 
-        #endregion
+        [Test]
+        public async Task CanDoHashExpire()
+        {
+            using var redis = ConnectionMultiplexer.Connect(TestUtils.GetConfig());
+            var db = redis.GetDatabase(0);
+            db.HashSet("myhash", [new HashEntry("field1", "hello"), new HashEntry("field2", "world"), new HashEntry("field3", "new")]);
+
+            var result = db.Execute("HEXPIRE", "myhash", "2", "FIELDS", "2", "field1", "field2");
+            var results = (RedisResult[])result;
+            ClassicAssert.AreEqual(2, results.Length);
+            ClassicAssert.AreEqual(1, (long)results[0]); // field1 success
+            ClassicAssert.AreEqual(1, (long)results[1]); // field2 success
+
+            var ttl = (RedisResult[])db.Execute("HTTL", "myhash", "FIELDS", "2", "field1", "field2");
+            ClassicAssert.AreEqual(2, ttl.Length);
+            ClassicAssert.LessOrEqual((long)ttl[0], 10);
+            ClassicAssert.Greater((long)ttl[0], 0);
+            ClassicAssert.LessOrEqual((long)ttl[1], 10);
+            ClassicAssert.Greater((long)ttl[1], 0);
+
+            await Task.Delay(2000);
+
+            var items = db.HashGetAll("myhash");
+            ClassicAssert.AreEqual(1, items.Length);
+            ClassicAssert.AreEqual("field3", items[0].Name.ToString());
+            ClassicAssert.AreEqual("new", items[0].Value.ToString());
+        }
+
+        [Test]
+        [TestCase("NX", Description = "Set expiry only when no expiration exists")]
+        [TestCase("XX", Description = "Set expiry only when expiration exists")]
+        [TestCase("GT", Description = "Set expiry only when new TTL is greater")]
+        [TestCase("LT", Description = "Set expiry only when new TTL is less")]
+        public void CanDoHashExpireWithOptions(string option)
```

Comment:
We need to run variants of the tests when the HASH has spilled to disk as well, in order to exercise those serialization etc. paths. See an example here: https://github.com/microsoft/garnet/blob/main/test/Garnet.test/ReadCacheTests.cs#L118